src/index.tsx

```diff
@@ -1,10 +1,14 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import { App } from "./components/app";
-
 import "./index.scss";
 
-ReactDOM.render(
-  <App />,
-  document.getElementById("app")
-);
+
+const render = () => {
+    ReactDOM.render(
+        <App />,
+        document.getElementById("app")
+    );
+};
+
+window.setTimeout(render, 2000);
```

Comment:
What happens without this?
Does QP render then codap initializes, and then the diagram changes to the saved version?
Or is there some bug that happens instead?

Yes the loading of the state in the component is temporary. I wanted to see what needs CODAP would have. And I was also hoping to integrate this into CLUE to see what I had to change to do that.

Reply:
QP loads the default version then switches to the saved version. I put in the delay to avoid the flash.